map-utils.js

```diff
@@ -1,126 +1,140 @@
+  var y = TILE_SIZE * (0.5 - Math.log((1 + siny) / (1 - siny)) / (4 * Math.PI));
+  if (opts.asPlainObject) {
+    return { x: x, y: y };
+  };
+
+  return new google.maps.Point(x, y);
+};
 var getLatLngFromPoint = function (point) {
-var lng = (point.x - tilePixelCenter) / pixelsPerLngDegree;
-var latRadians = (point.y - tilePixelCenter) / -pixelsPerLngRadian; 
-var lat = radiansToDegrees(2 * Math.atan(Math.exp(latRadians)) - Math.PI / 2);
-return {lat: lat, lng: lng} 
-}
+  var lng = (point.x - tilePixelCenter) / pixelsPerLngDegree;
+  var latRadians = (point.y - tilePixelCenter) / -pixelsPerLngRadian;
+  var lat = radiansToDegrees(2 * Math.atan(Math.exp(latRadians)) - Math.PI / 2);
+
+  return { lat: lat, lng: lng };
+};
 var getFormattedPlaceName = function (placeObj) {
-if (!placeObj.formatted_address) {
-return null
-}
-var htmlAddress = placeObj.adr_address
-var firstPart = htmlAddress.substr(0, htmlAddress.indexOf("<"))
-var addressContainer = window.document.createElement("span")
-addressContainer.innerHTML = htmlAddress
-var childNodes = Array.prototype.slice.call(addressContainer.children, 0)
-if (childNodes.length == 0) {
-return placeObj.formatted_address
-}
-return firstPart + childNodes
-  .filter(function (node) { return node.className != "postal-code"})
-  .map(function (node) { return node.textContent})
-  .join(", ")  
-}
+  if (!placeObj.formatted_address) {
+    return null;
+  };
+  var htmlAddress = placeObj.adr_address;
+  var firstPart = htmlAddress.substr(0, htmlAddress.indexOf('<'));
+  var addressContainer = window.document.createElement('span');
+  addressContainer.innerHTML = htmlAddress;
+  var childNodes = Array.prototype.slice.call(addressContainer.children, 0);
+  if (childNodes.length === 0) {
+    return placeObj.formatted_address;
+  };
+
+  return firstPart + childNodes
+    .filter(function (node) { return node.className !== 'postal-code';})
+    .map(function (node) { return node.textContent;})
+    .join(', ');
+};
 var getAdjustedMapCenter = function (options) {
-var mapCenter = options.mapCenter;
-var mapZoom = options.mapZoom;
-var offset = options.offset;
-var reverse = options.reverse;
-var directionalOffset = reverse ? offset * -1 : offset
-z = Math.pow(2, mapZoom)
-console.log('z: ', z);
-var point = getPointFromLatLng({
-lat: result(mapCenter, "lat"),
-lng: result(mapCenter, "lng"),
-}, {asPlainObject: true})
-var adjustedMapCenter = getLatLngFromPoint({
-x: point.x,
-y: point.y + (directionalOffset / 2) / z,
-})
-adjustedMapCenter.lat = round(adjustedMapCenter.lat, 6)
-adjustedMapCenter.lng = round(adjustedMapCenter.lng, 6)
-return adjustedMapCenter      
-}
+  var mapCenter = options.mapCenter;
+  var mapZoom = options.mapZoom;
+  var offset = options.offset;
+  var reverse = options.reverse;
+  var directionalOffset = reverse ? offset * -1 : offset;
+  var z = Math.pow(2, mapZoom);
+  //console.log('z: ', z);
+  var point = getPointFromLatLng({
+    lat: mapCenter.lat,
```

Comment:
This qualitatively changes the meaning of the code, but this was a tricky part of the assignment. The original usage of `result()` was to evaluate the method specified on the object specified - essentially `mapCenter.lat()` and `mapCenter.lng()`. This was a method originally imported from lodash through the use of a require statement. To work around the issue you could add the method to globals (preferred since this is technically what it is) or a var declaration (not preferred since during compilation the JavaScript engine would simply ignore the re-declaration.)